bbq/vm/config.go

```diff
@@ -102,6 +111,107 @@ func (c *Config) Interpreter() *interpreter.Interpreter {
 	return c.inter
 }
 
+func (c *Config) MeterMemory(usage common.MemoryUsage) error {
+	if c.MemoryGauge == nil {
+		return nil
+	}
+
+	return c.MemoryGauge.MeterMemory(usage)
+}
+
+func (c *Config) TrackReferencedResourceKindedValue(
+	id atree.ValueID,
+	value *EphemeralReferenceValue,
+) {
+	values := c.referencedResourceKindedValues[id]
+	if values == nil {
+		values = map[*EphemeralReferenceValue]struct{}{}
+		c.referencedResourceKindedValues[id] = values
+	}
+	values[value] = struct{}{}
+}
+
+func (c *Config) ReferencedResourceKindedValues(valueID atree.ValueID) map[*EphemeralReferenceValue]struct{} {
+	return c.referencedResourceKindedValues[valueID]
+}
+
+func (c *Config) ClearReferenceTracking(valueID atree.ValueID) {
+	delete(c.referencedResourceKindedValues, valueID)
+}
+
+func (c *Config) ReadStored(storageAddress common.Address, domain common.StorageDomain, identifier interpreter.StorageMapKey) interpreter.Value {
+	accountStorage := c.GetDomainStorageMap(
+		c.Interpreter(),
+		storageAddress,
+		domain,
+		false,
+	)
+	if accountStorage == nil {
+		return nil
+	}
+
+	return accountStorage.ReadValue(c, identifier)
+}
+
+func (c *Config) WriteStored(
+	storageAddress common.Address,
+	domain common.StorageDomain,
+	key interpreter.StorageMapKey,
+	value interpreter.Value,
+) (existed bool) {
+	inter := c.Interpreter()
+	accountStorage := c.GetDomainStorageMap(inter, storageAddress, domain, true)
+
+	return accountStorage.WriteValue(
+		inter,
+		key,
+		value,
+	)
+}
+
+func (c *Config) ConvertStaticToSemaType(staticType interpreter.StaticType) (sema.Type, error) {
+	inter := c.Interpreter()
+	return inter.ConvertStaticToSemaType(staticType)
+}
+
+func (c *Config) IsSubType(subType interpreter.StaticType, superType interpreter.StaticType) bool {
+	inter := c.Interpreter()
+	return inter.IsSubType(subType, superType)
+}
+
+func (c *Config) IsSubTypeOfSemaType(staticSubType interpreter.StaticType, superType sema.Type) bool {
+	inter := c.Interpreter()
+	return inter.IsSubTypeOfSemaType(staticSubType, superType)
+}
```

Comment:
These functions are later removed (in follow-up PRs), as the interpreter functions were re-used.  So this is kind of temporary.